mlflow/store/sqlalchemy_store.py

```diff
@@ -184,48 +186,68 @@ def log_metric(self, run_uuid, metric):
             raise MlflowException('Metric={} must be unique'.format(metric),
                                   INVALID_PARAMETER_VALUE)
 
+    def get_metric(self, run_uuid, metric_key):
+        metric = self.session.query(SqlMetric).filter_by(
+            run_uuid=run_uuid, key=metric_key
+        ).order_by(sqlalchemy.desc(SqlMetric.timestamp)).first()
+
+        if metric is None:
+            raise MlflowException('Metric={} does not exist'.format(metric_key),
+                                  RESOURCE_DOES_NOT_EXIST)
+
+        return metric.to_mlflow_entity()
+
+    def get_metric_history(self, run_uuid, metric_key):
+        metrics = self.session.query(SqlMetric).filter_by(run_uuid=run_uuid, key=metric_key).all()
+        return [metric.to_mlflow_entity() for metric in metrics]
+
+    def get_all_metrics(self, run_uuid):
```

Comment:
As discussed offline, would be good to remove these currently-unnecessary methods (get_all_metrics/get_all_params/get_all_tags).